### ISSUE TITLE
A runtime bug

### ISSUE THREAD

**Issue by YKDZ** [vikejs/vike#1912]
### Description

```
[vike@0.4.197][Bug] You stumbled upon a Vike bug. Go to https://github.com/vikejs/vike/issues/new and copy-paste this error. A maintainer will fix the bug (usually under 24 hours).
    at LoadPluginContext.load (file:///C:/Users/YKDZ/Desktop/Yggdrasil/node_modules/.pnpm/vike@0.4.197_vite@5.4.8_@types+node@22.7.0_/node_modules/vike/dist/esm/node/plugin/plugins/fileEnv.js:21:13)
    at PluginContainer.load (file:///C:/Users/YKDZ/Desktop/Yggdrasil/node_modules/.pnpm/vite@5.4.8_@types+node@22.7.0/node_modules/vite/dist/node/chunks/dep-CDnG8rE7.js:49002:17)
    at loadAndTransform (file:///C:/Users/YKDZ/Desktop/Yggdrasil/node_modules/.pnpm/vite@5.4.8_@types+node@22.7.0/node_modules/vite/dist/node/chunks/dep-CDnG8rE7.js:51807:22)
    at viteTransformMiddleware (file:///C:/Users/YKDZ/Desktop/Yggdrasil/node_modules/.pnpm/vite@5.4.8_@types+node@22.7.0/node_modules/vite/dist/node/chunks/dep-CDnG8rE7.js:61824:24)
```

**Comment by brillout** [comment]
Insert the following before line `21` of `C:/Users/YKDZ/Desktop/Yggdrasil/node_modules/.pnpm/vike@0.4.197_vite@5.4.8_@types+node@22.7.0_/node_modules/vike/dist/esm/node/plugin/plugins/fileEnv.js` and report back the logs.

```diff
             const moduleInfo = viteDevServer.moduleGraph.getModuleById(id);
+            if (!moduleInfo) console.log('id', id)
             assert(moduleInfo);
```

**Comment by YKDZ** [comment]
> Insert the following before line `21` of `C:/Users/YKDZ/Desktop/Yggdrasil/node_modules/.pnpm/vike@0.4.197_vite@5.4.8_@types+node@22.7.0_/node_modules/vike/dist/esm/node/plugin/plugins/fileEnv.js` and report back the logs.
> 
> ```diff
>              const moduleInfo = viteDevServer.moduleGraph.getModuleById(id);
> +            if (!moduleInfo) console.log('id', id)
>              assert(moduleInfo);
> ```

```
id C:/Users/YKDZ/Desktop/Yggdrasil/src/components/MailMessageInbox.client.vue?vue&type=style&index=0&scoped=037115ff&lang.css?direct
16:33:50 [vike@0.4.197][Bug] You stumbled upon a Vike bug. Go to https://github.com/vikejs/vike/issues/new and copy-paste this error. A maintainer will fix the bug (usually under 24 hours).
    at LoadPluginContext.load (file:///C:/Users/YKDZ/Desktop/Yggdrasil/node_modules/.pnpm/vike@0.4.197_vite@5.4.8_@types+node@22.7.0_/node_modules/vike/dist/esm/node/plugin/plugins/fileEnv.js:22:13)
    at PluginContainer.load (file:///C:/Users/YKDZ/Desktop/Yggdrasil/node_modules/.pnpm/vite@5.4.8_@types+node@22.7.0/node_modules/vite/dist/node/chunks/dep-CDnG8rE7.js:49002:17)
    at loadAndTransform (file:///C:/Users/YKDZ/Desktop/Yggdrasil/node_modules/.pnpm/vite@5.4.8_@types+node@22.7.0/node_modules/vite/dist/node/chunks/dep-CDnG8rE7.js:51807:22)

```

**Comment by brillout** [comment]
Does it work with this?

```diff
+            if (id.endsWith('?direct')) id = id.slice(0, -1 * '?direct'.length))
             const moduleInfo = viteDevServer.moduleGraph.getModuleById(id);
+            if (!moduleInfo) console.log('id', id)
             assert(moduleInfo);
```

**Comment by YKDZ** [comment]
> Does it work with this?
> 
> ```diff
> +            if (id.endsWith('?direct')) id = id.slice(0, -1 * '?direct'.length))
>              const moduleInfo = viteDevServer.moduleGraph.getModuleById(id);
> +            if (!moduleInfo) console.log('id', id)
>              assert(moduleInfo);
> ```

Yes, now it works well.

**Comment by brillout** [comment]
Fix pre-released as `0.4.198-commit-081c85f`.
